tests/rest/RestIssueTest.php

```diff
@@ -547,6 +547,21 @@ public function testCreateIssueNoCategory() {
 		}
 	}
 
+	public function testCreateIssueWithCategoryInvalid() {
+		$t_issue_to_add = $this->getIssueToAdd();
+		$t_issue_to_add['category'] = 'Non-existent category';
+
+		$t_response = $this->builder()->post( '/issues', $t_issue_to_add )->send();
+
+		$this->assertEquals( HTTP_STATUS_NOT_FOUND, $t_response->getStatusCode() );
```

Comment:
I wonder if in such case we should be returning NOT FOUND or BAD REQUEST.

- NOT FOUND - the argument is that format is valid, just one of the foreign keys in not found.
- BAD REQUEST - the argument is that 404 should be returned if we are attempting to get an element that doesn't exist, but in this case we are creating an element with an invalid value, hence, bad request.